src/ng/rootScope.js

```diff
@@ -1350,6 +1311,24 @@ function $RootScopeProvider() {
 
     return $rootScope;
 
+    function invokeListeners(scope, event, listenerArgs, name) {
+      var listeners = scope.$$listeners[name];
+      if (listeners) {
+        if (listeners.$$index !== -1) {
+          throw $rootScopeMinErr('inevt', '{0} already $emit/$broadcast-ing', name);
```

Comment:
Maybe worth mentioning the scope `$id` (or even whether it was being `$emit`ed/`$broadcast`ed) to aid in debugging).

Also, we need an error page for `$rootScope:inevt`.